src/sentry/integrations/source_code_management/repository.py

```diff
@@ -177,26 +181,22 @@ def get_codeowner_file(
          * filepath - full path of the file i.e. CODEOWNERS, .github/CODEOWNERS, docs/CODEOWNERS
          * raw - the decoded raw contents of the codeowner file
         """
-        if self.codeowners_locations is None:
-            raise NotImplementedError("Implement self.codeowners_locations to use this method.")
-
-        for filepath in self.codeowners_locations:
-            html_url = self.check_file(repo, filepath, ref)
-            if html_url:
-                try:
-                    contents = self.get_client().get_file(repo, filepath, ref, codeowners=True)
-                    metrics.incr(
-                        REPOSITORY_INTEGRATION_GET_FILE_METRIC.format(result="success"),
-                        tags={"integration": self.integration_name},
-                    )
-                except ApiError:
-                    metrics.incr(
-                        REPOSITORY_INTEGRATION_GET_FILE_METRIC.format(result="success"),
-                        tags={"integration": self.integration_name},
-                    )
-                    continue
-                return {"filepath": filepath, "html_url": html_url, "raw": contents}
-        return None
+        with self.record_event(
+            RepositoryIntegrationInteractionType.GET_CODEOWNER_FILE
+        ).capture() as lifecycle:
+            if self.codeowners_locations is None:
+                raise NotImplementedError("Implement self.codeowners_locations to use this method.")
+
+            for filepath in self.codeowners_locations:
+                html_url = self.check_file(repo, filepath, ref)
+                if html_url:
+                    try:
+                        contents = self.get_client().get_file(repo, filepath, ref, codeowners=True)
+                    except ApiError:
+                        lifecycle.record_failure(ApiError)
```

Comment:
i don't think we need to emit an error here either, it seems to be an acceptable error? again, i'm not sure if returning `None` should be counted as an error... we return early if we find something though